pages/blog/building-async-flight-notification-service.md

```diff
@@ -0,0 +1,249 @@
+            "terminal": {  
+                "code": "1"  
+            },  
+            "gate": {  
+                "mainGate": "A04"  
+            },  
+            "timings": [  
+                {  
+                    "qualifier": "STA",  
+                    "value": "2020-11-05T19:35+01:00"  
+                }  
+            ]  
+        }  
+    }
+]
+```
+
+We can see that:  
+
+- The flight is scheduled to depart from Terminal 1, Gate B22 of Frankfurt International Airport (FRA) at 18:20 (UTC+1). 
+- It is scheduled to arrive at Terminal 1, Gate A04 of Amsterdam Schiphol Airport (AMS) at 19:35 (UTC+1). 
+
+The API is synchronous and therefore needs to be polled to monitor the flight status. This isn’t ideal and we need a solid strategy to avoid DDoSing the Amadeus backend, using up our free call quota or piling up a massive bill at the end of the month. 
+
+To solve this, we put the Monitor service on a separate thread. Every five minutes, the thread checks to see if it’s time to retrieve information from the API and update the status. The Monitor only calls the API if two conditions are met: 
+
+- The current date is equal to the departure date. 
+- The current time is within 4 hours of the departure time. 
+
+## Subscribing to flight updates 
+
+The Subscriber service  lets users subscribe to the notifications. We built a simple HTTP server with Flask to let the user enter their name, phone number and flight information.
+
+Once it receives this information, the Subscriber emits a `flightQueue` message with that information in the payload to the broker, so that it can be received by the Monitor.
+
+## Sending notifications to users 
+
+The Notifier service receives flight status updates from the Monitor and uses the Twilio SMS API to notify the end. The service has a very simple implementation: when the Notifier receives a `flightStatus` message, it uses the message payload to build an SMS message:
+
+```python
+client = twilio.Client(account_sid, auth_token)
+
+msg = build_message(alert_msg['user'],
+                    alert_msg['departure'],
+                    alert_msg['arrival'])
+
+destination_phone = alert_msg['user']['phoneNumber']
+
+message = client.messages.create(body=msg,
+                                 from_=twilio_phone,
+                                 to=destination_phone)
+```
+
+## Running the service 
+
+The prototype runs on four Docker containers – one per service plus another for the [MQTT broker](https://github.com/toke/docker-mosquitto) based on the Docker image maintained by the Eclipse Mosquitto project.
+
+To avoid manually starting each service (plus the dependency of starting the broker first), we will use [Docker compose](https://docs.docker.com/compose/), a tool to run applications composed of multiple containers using a YAML file to define each container as well as their dependencies.
+
+We start the service by executing:
+
+        docker-compose up --remove-orphans 
```

Comment:
I had to do `docker network create my-network` first

also, good to mention link to project https://github.com/amadeus4dev/amadeus-async-flight-status in this section too as I had to go to the end of the post to actually find it